src/js/components/pages/tag-editor.jsx

```diff
@@ -123,21 +127,53 @@ var TagEditor = React.createClass({
                     //add tags as long as they are not empty
                     if (mediaObjectTags[i] != "") {
                         tags.push({id: i, text: mediaObjectTags[i].trim()})
+                        textOnlyTags.push(mediaObjectTags[i].trim())
                     }
 
                 }
             }
+
+            //get any suggested ReactTags_blank
+            var suggestedTags = [];
+            var self = this;
+            this.getRemoteTags(mediaObject, function (remoteTags) {
+              //check each suggested tag isn't already in the scene tags
+              for (var i =0; i < remoteTags.length; i++) {
+                if(textOnlyTags.indexOf(remoteTags[i]) === -1) {
+                    suggestedTags.push({id: i, text: remoteTags[i]})
+                }
+              }
+              self.setState({tags2: suggestedTags})
+              self.forceUpdate() //don't know why but dosn't work without this.
+            })
+
         }
-        //update display
+
+        //update display - potentialy race condition but should always happen before remote responds
         this.setState({
             scene: scene,
             suggestions: this.getSceneTags(scene),
             focusedMediaObject: focusedMediaObject,
-            tags: tags
+            tags: tags,
+            tags2: []
         });
 
     },
 
+    //fetch tags as text only from image processing service
+    getRemoteTags(mediaObject, callback) {
```

Comment:
Would be nice to move this into a more generic helper, something like SceneActions but maybe AssetActions?

This would in future allow us to easily pull tags in other workflows.  I imagine it is a small change too.
Could potentially look at caching these remote tag results in the frontend (React Store too), given we cache in the backend, it's not such an issue.  Could still be nice just to encapsulate these values in something that lives longer than the component itself.

Reply:
Agree with pulling it out to separate Actions helper (this can eventually handle other actions like getting color pallet or find features of interest or matching objects if we decide to integrate those sort of workflows). Also makes it reusable for tagging on upload.

I would argue against caching in the front end for now because if the image processing service expands to include some of the more interesting features then we might want to modify tags on the back-end and therefore the cached state in the UI would become invalid.

Reply:
Sure thing, more than happy to leave out the frontend cache, especially as it's properly cached in the backend.  It's easy enough to add in the future if necessary.